### ISSUE TITLE
findChromPeaks() does not seem to release memory 

### ISSUE THREAD

**Issue by maiyama** [sneumann/xcms#355]
Hello, 
I have been having an issue with findChromPeaks() function with 8 mzML files (150MB each, containing data of 35min. retention time) when running codes on 16GB RAM Linux Ubuntu 18.04 system with 8 cores. Memory does not seem to be released when running findChromPeaks() and system freezes. It works when running 6 files instead of 8, which indicates that all the files are being loaded, taking up the memory and not being released (i.e. not being read one at the time). R version is 3.5.2 and I am using xcms version 3.4.4. 
My issue seems related to the past issue that I linked to this new issue. 
I hope to get advice on this situation to move past this step with larger data that I have (60 samples).  Thanks. 

#################################################

Hi Karl,

I made a small change that could eventually help to reduce memory demand a bit. Could you please try it?

To install:
```r
library(devtools)
install_github("sneumann/xcms", ref = "RELEASE_3_6")
```

Let me know if this installation procedure does not work for you - I will then try to compile the package for Windows.

_Originally posted by @jotsetung in https://github.com/sneumann/xcms/issues/256#issuecomment-369499446_

**Comment by jotsetung** [comment]
Dear @maiyama , I guess that your data is too big to be loaded completely into memory. `xcms` will load the data from one file into memory for each parallel processing task/thread. If you run your code on 8 cores, the data from 8 files will have to fit into memory. Note also that `R` needs a little more memory to be able to copy and merge the results from each process.

To know the actual memory size required for one file you could to the following (assuming `data` is the object you got with `readMSData`):

```r
pryr::object_size(spectra(filterFile(data, 1)))
```

**Comment by maiyama** [comment]
Dear @jotsetung , Thank you for the quick response and helpful advice. 
After running the 
`pryr::object_size(spectra(filterFile(data, 1)))`

it indicated that one of my files require 205 MB to process, which seems quite large. I tried with much smaller data (15MB each, instead of 150MB) and it took only 6 seconds to go through findChromPeaks() for 3 files. 
I also tried using 2 cores instead of 6 (default) with the large files that previously froze my computer and it worked this time! 
So it seems like the issue was because of both the large file size and multi-core processing as you suggested.
